recipes/hadoop_yarn_nodemanager.rb

```diff
@@ -52,7 +52,14 @@
 end
 
 # Ensure permissions for secure Hadoop... this *should* be no-op
-file '/usr/lib/hadoop-yarn/bin/container-executor' do
+container_executor_path =
+  if node['hadoop']['distribution'] == 'hdp' && node['hadoop']['distribution_version'].to_f == 2.2
```

Comment:
This needs an additional check. Using `2` also is HDP 2.2...

https://github.com/caskdata/hadoop_cookbook/blob/master/recipes/repo.rb#L59-L61

Do **not** use `.to_i` here, use `.to_s` and check for the string `'2'`